### DISCUSSION TITLE
Is there any way to use hashes in a .mailmap file rather than name/emails?

### DISCUSSION THREAD

**Discussion by eamodio** [git-for-windows/git#3072] [Q&A] [answered]
I'm guessing not from looking at the [docs](https://git-scm.com/docs/git-shortlog#_mapping_authors), but this would be a great privacy boost IMO.

For example, on the VS Code team, I want to set up a `.mailmap` for the repo, since we have so many members who commit using multiple different names/emails. But, many team members don't want their email addresses exposed in a simple place like that. Yes, the emails are accessible in the repo, but its so much easier to run a script on GitHub/etc and collect `.mailmap` files to grab a bunch of emails at once.

So if there was a way where we could use a hash of the user/email combo (using sha1 or something) rather than the user/email directly and Git could match against that. It would help improve that privacy.

**Comment by rimrul** [comment, marked as answer]
There has been [some work towards such a feature](https://lore.kernel.org/git/20210103211849.2691287-1-sandals@crustytoothpaste.net/), but I think it's not possible in any released versions (or the upcoming version 2.31.0) of git.

**Reply by eamodio** [reply]
Thanks!
